pkg/controllers/manageagent/manageagent.go

```diff
@@ -205,6 +207,7 @@ func (h *handler) newAgentBundle(ns string, cluster *fleet.Cluster) (runtime.Obj
 				DefaultNamespace: agentNamespace,
 				Helm: &fleet.HelmOptions{
 					TakeOwnership: true,
+					ReleaseName:   name2.Limit(name.SafeConcatName(AgentBundleName, cluster.Name), 53),
```

Comment:
why are we not using the new `HelmReleaseName` func? Or maybe we should not add `ReleaseName` here. Then `ReleaseName` would be nil and `HelmReleaseName` would be called in [getOpts](https://github.com/rancher/fleet/blob/master/pkg/helmdeployer/deployer.go#L258-L261) when installing the helm release.

Reply:
Thanks, that's a good point. This was my initial fix, before I noticed we derive the release name differently in apply and deployer. But with the change to deployer, this should not be needed anymore.